service/worker/scanner/build_ids/scavenger.go

```diff
@@ -210,7 +211,12 @@ func (a *Activities) processNamespaceEntry(
 				return ctx.Err()
 			}
 			entry := tqResponse.Entries[heartbeat.TaskQueueIdx]
-			if err := a.processUserDataEntry(ctx, rateLimiter, *heartbeat, ns, entry); err != nil {
+			err := a.processUserDataEntry(ctx, rateLimiter, *heartbeat, ns, entry)
+			if common.IsContextDeadlineExceededErr(err) {
```

Comment:
nit: you can put this condition inside the `err != nil` block and keep the old structure.

Can you check for `errors.Is(err, errNonRetryable) || ctx.Err() != nil` instead?